src/Model/MaterialModel_GOH14_ST91_Mixed.cpp

```diff
@@ -0,0 +1,278 @@
+    h5w -> write_doubleScalar("f2_phi", f2_phi);
+    h5w -> write_doubleScalar("fk1", fk1);
+    h5w -> write_doubleScalar("fk2", fk2);
+    h5w -> write_doubleScalar("fkd", fkd);
+
+    delete h5w; H5Fclose(file_id);
+  }
+
+  MPI_Barrier(PETSC_COMM_WORLD);
+}
+
+void MaterialModel_GOH14_ST91_Mixed::get_PK( 
+    const Tensor2_3D &F, Tensor2_3D &P, Tensor2_3D &S) const
+{
+  Tensor2_3D C; C.MatMultTransposeLeft(F);
+  Tensor2_3D Cinv = Ten2::inverse(C);
+  const double trC = C.tr();
+  const double detFm0d67 = std::pow(F.det(), mpt67);
+
+  const double a1Ca1 = C.VecMatVec(a1, a1);
+  const double a2Ca2 = C.VecMatVec(a2, a2);
+
+  const double fE1 = fkd*trC + (1.0-3.0*fkd)*a1Ca1 - 1.0;
+  const double fE2 = fkd*trC + (1.0-3.0*fkd)*a2Ca2 - 1.0;
+  
+  const double dfpsi1 = fk1 * fE1 * std::exp( fk2 * fE1 * fE1 );
+  const double dfpsi2 = fk1 * fE2 * std::exp( fk2 * fE2 * fE2 );
+
+  // P : I = I - 1/3 trC C^-1
+  Tensor2_3D PxI(Cinv); PxI.scale( (-1.0) * pt33 * trC );
+  PxI.AXPI( 1.0 );
+
+  Tensor2_3D PxH1; Tensor2_3D PxH2;
+  PxH1.gen_zero(); PxH2.gen_zero();
+  PxH1.AXPI( fkd );
+  PxH2.AXPI( fkd );
+
+  Tensor2_3D a1xa1, a2xa2;
+  a1xa1.gen_outprod(a1);
+  a2xa2.gen_outprod(a2);
+
+  PxH1.AXPY( 1.0-3.0*fkd, a1xa1 );
+  PxH2.AXPY( 1.0-3.0*fkd, a2xa2 );
+
+  S.gen_zero();
+  S.AXPY( mu * detFm0d67, PxI);
+  
+  S.AXPY(2.0 * dfpsi1, PxH1);
+  S.AXPY(2.0 * dfpsi2, PxH2);
+
+  // 1st PK
+  P.MatMult(F,S);
+}
+
+void MaterialModel_GOH14_ST91_Mixed::get_PK_Stiffness( 
+    const Tensor2_3D &F, Tensor2_3D &P, Tensor2_3D &S, Tensor4_3D &CC ) const
+{
+  Tensor2_3D C; C.MatMultTransposeLeft(F);
+  Tensor2_3D Cinv = Ten2::inverse(C);
+  const double trC = C.tr();
+  const double detFm0d67 = std::pow(F.det(), mpt67);
+
+  const double a1Ca1 = C.VecMatVec(a1, a1);
+  const double a2Ca2 = C.VecMatVec(a2, a2);
+
+  const double fE1 = fkd*trC + (1.0-3.0*fkd)*a1Ca1 - 1.0;
+  const double fE2 = fkd*trC + (1.0-3.0*fkd)*a2Ca2 - 1.0;
+  
+  const double dfpsi1 = fk1 * fE1 * std::exp( fk2 * fE1 * fE1 );
+  const double dfpsi2 = fk1 * fE2 * std::exp( fk2 * fE2 * fE2 );
+
+  // P : I = I - 1/3 trC C^-1
+  Tensor2_3D PxI(Cinv); PxI.scale( (-1.0) * pt33 * trC );
+  PxI.AXPI( 1.0 );
+
+  Tensor2_3D PxH1; Tensor2_3D PxH2;
+  PxH1.gen_zero(); PxH2.gen_zero();
+  PxH1.AXPI( fkd );
+  PxH2.AXPI( fkd );
+
+  Tensor2_3D a1xa1, a2xa2;
+  a1xa1.gen_outprod(a1);
+  a2xa2.gen_outprod(a2);
+
+  PxH1.AXPY( 1.0-3.0*fkd, a1xa1 );
+  PxH2.AXPY( 1.0-3.0*fkd, a2xa2 );
+  
+  S.gen_zero();
+  S.AXPY( mu * detFm0d67, PxI);
+  
+  S.AXPY(2.0 * dfpsi1, PxH1);
+  S.AXPY(2.0 * dfpsi2, PxH2);
+  
+  P.MatMult(F,S);
+
+  // Assembly the stiffness matrix
+  const double d2fpsi1 = fk1 * (1.0 + 2.0*fk2*fE1*fE1) * std::exp(fk2*fE1*fE1); 
+  const double d2fpsi2 = fk1 * (1.0 + 2.0*fk2*fE2*fE2) * std::exp(fk2*fE2*fE2); 
+
+  CC.gen_zero();
+
+  CC.add_OutProduct(4.0 * d2fpsi1, PxH1, PxH1);
+  CC.add_OutProduct(4.0 * d2fpsi2, PxH2, PxH2);
+
+  const double val = mpt67 * mu * detFm0d67;
+
```

Comment:
For the isotropic elasticity tensor, I am not sure the result is identical to the one of the NeoHookean model.  A check or a discussion with me is needed.